src/popup/components/TransactionOverview.vue

```diff
@@ -8,103 +8,131 @@
   />
 </template>
 
-<script>
-import { mapState, mapGetters } from 'vuex';
+<script lang="ts">
 import { SCHEMA } from '@aeternity/aepp-sdk';
-import { watchUntilTruthy } from '../utils';
+import {
+  computed, defineComponent, onMounted, ref,
+} from '@vue/composition-api';
+import {
+  TRANSACTION_OWNERSHIP_STATUS,
+  TRANSACTION_DIRECTION_RECEIVED,
+  TRANSACTION_DIRECTION_SENT,
+  watchUntilTruthy,
+} from '../utils';
 import TransactionInfo from './TransactionInfo.vue';
+import { useGetter, useState, useTransaction } from '../../composables';
+import { ITransaction, TransactionType } from '../../types';
 
-export default {
+export default defineComponent({
   components: { TransactionInfo },
   props: {
     tx: { type: Object, required: true },
-    isDex: { type: Boolean, default: false },
   },
-  data: () => ({ name: '' }),
-  computed: {
-    ...mapGetters(['getTxType', 'getTxDirection', 'getExplorerPath', 'getDexContracts']),
-    ...mapGetters('names', ['getPreferred']),
-    ...mapState({
-      account(_, { account }) {
-        return {
-          ...account,
-          label: this.$t('transaction.overview.accountAddress'),
-          url: this.getExplorerPath(account.address),
-        };
-      },
-      isDexRecipient() {
-        return [...this.getDexContracts.router, ...this.getDexContracts.wae]
-          .includes(this.tx?.contractId);
-      },
-    }),
-    transaction() {
-      switch (this.txType) {
+  setup(props, { root }) {
+    const name = ref('');
+
+    const middleware = useState('middleware');
+
+    const getTxDirection = useGetter('getTxDirection');
+    const getExplorerPath = useGetter('getExplorerPath');
+    const getDexContracts = useGetter('getDexContracts');
+    const getPreferred = useGetter('names/getPreferred');
+
+    const {
+      ownershipStatus,
+      ownershipAccount,
+      txType,
+      setTransaction,
+    } = useTransaction();
+
+    setTransaction({ tx: props.tx } as ITransaction);
+
+    const isDexRecipient = computed(
+      () => [
+        ...getDexContracts.value.router,
+        ...getDexContracts.value.wae,
+      ].includes(props.tx?.contractId),
+    );
+    const transaction = computed(() => {
+      const transactionTypes = root.$t('transaction.type') as Record<TransactionType, any>;
```

Comment:
must be any?

Reply:
you set that type, but it should be changed to `TranslateResult`